nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -45,12 +50,14 @@ let
       ipfs --local config Mounts.IPNS ${cfg.ipnsMountDir}
     '' + concatStringsSep "\n" (collect
           isString
-          (mapAttrsRecursive
-            (path: value:
+          (mapAttrsRecursiveCond (value: ! value ? _type)
+            (path: value: let
+              value' = if value ? _type then value.content else value;
```

Comment:
Why is this suddenly needed?

Reply:
Attrsets in configuration.nix evaluate to these kind of attrset wrappers (for purposes of handling `mkOverride` IIUC), rather than the attrsets literally provided. I had not sufficiently tested using attrsets as values, so I only recently found this change was necessary for proper marshalling.

Reply:
I don't get it, can you add a comment explaining this with an example of what wouldn't work without it?

Reply:
`services.ipfs.extraConfig.Swarm.AddrFilters = [];` wouldn't work right now in master, because the attrset `Swarm` is not flattened into a simple attrset at the time the recursive map is evaluated. I can't explain why, because I don't understand it myself. I only know that the attrset I'm expecting is in `.contents` whenever `.type` is present.

Reply:
Can't reproduce, with or without this change, the mentioned setting works without problems.

Reply:
Update on this?